### PR TITLE
Feature/677 backend

### PR DESCRIPTION
**Pull request by Palacios95**
Added group logic and router for back end.

### CODE REVIEW

File: controllers/groupController.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+var crypto = require('crypto');
+var group_socket = require('../server');
+
+exports.create_group = (req, res) =>  {
+    //Randomly generated 5 bytes. Encoded in base64.
+    var uid = crypto.randomBytes(4).toString('hex');
```
Comment (arodr967) [inline controllers/groupController.js:6]: **Best Practice:** Don't use `var`, use `const` or `let`.

File: routes/group.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+var express = require('express');
+var router = express.Router();
+
+var group_controller = require('../controllers/groupController');
```
Comment (arodr967) [inline routes/group.js:4]: **Best Practice**: Don't use `var`, use `const` or `let`.

File: server.js [hunk 3]
```diff
@@ -5,10 +5,15 @@ const app = express();
 const cors = require('cors');
 const yelp = require('yelp-fusion');
 const Workfrom = require('workfrom');
+var http = require('http').Server(app);
+var server = require("http").Server(app);
+var io = require('socket.io').listen(server);
+var group_socket = require('./sockets/groupSocket')(io);
```
Comment (arodr967) [inline server.js]: **Best Practice**: Don't use `var`, use `const` or `let`.

File: sockets/groupSocket.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+                });
+
+                socket.on('sendUserInfo', (info) => {
+                    console.log(`Sending info to specific client: ${info.socketID}`)
+                    socket.to(info.socketID).emit('getExistingUserInfo', info.userInfo);
+                })
+
+                //Whenever someone disconnects this piece of code executed
+                socket.on('disconnect', function () {
+                    console.log('A user disconnected');
```
Comment (arodr967) [inline sockets/groupSocket.js:26]: So do they _actually_ get disconnected?

**Review by arodr967 — COMMENTED** [review]
Good job!

File: sockets/groupSocket.js (hunk 4 above)
Comment (Palacios95) [reply sockets/groupSocket.js:26]: When the app closes, they disconnect from the socket.

File: controllers/groupController.js (hunk 1 above)
Comment (Palacios95) [reply controllers/groupController.js:6]: fixed

File: routes/group.js (hunk 2 above)
Comment (Palacios95) [reply routes/group.js:4]: fixed

File: server.js (hunk 3 above)
Comment (Palacios95) [reply server.js]: fixed

**Review by arodr967 — APPROVED** [review]

**Pull request merged**